testing/test_config.py

```diff
@@ -1274,6 +1274,191 @@ def test_with_existing_file_in_subdir(self, tmpdir) -> None:
+                    print('\\nappended_option:%s\\n' % ini_val)""".format(
+                opt_name=ini_option_name
+            )
+        )
+
+        result = testdir.runpytest(
+            "--append-ini", "{}={}".format(ini_option_name, append_value), "-s"
+        )
+        result.stdout.fnmatch_lines(match_value)
+
+        result = testdir.runpytest(
+            "--append-ini",
+            "{}={}".format(ini_option_name, append_value),
+            "-s",
+            "--strict-config",
+        )
+
+    @pytest.mark.parametrize(
+        "ini_option_name, append_value, stdout_match, stderr_match",
+        [
+            (
+                "my_bool",
+                "False",
+                "appended_option:True",
+                "append_ini option invalid for argument 'my_bool' with type 'bool'",
+            ),
+            (
+                "my_none",
+                "None",
+                "appended_option:None",
+                "append_ini option invalid for argument 'my_none' with type 'None'",
+            ),
+        ],
+    )
+    def test_append_ini_usage_invalid_types(
+        self, testdir, ini_option_name, append_value, stdout_match, stderr_match
+    ):
+        testdir.makeconftest(
+            """
+            def pytest_addoption(parser):
+                addini = parser.addini
+                addini("my_bool", "", default=True, type="bool")
+                addini("my_none", "", default=None, type=None)"""
+        )
+        testdir.makeini(
+            """
+            [pytest]
+            my_bool=True
+            my_none=None
+        """
+        )
+        testdir.makepyfile(
+            """
+            def test_pass(pytestconfig):
+                ini_val = pytestconfig.getini("{opt_name}")
+                print('\\nappended_option:%s\\n' % ini_val)""".format(
+                opt_name=ini_option_name
+            )
+        )
+
+        result = testdir.runpytest(
+            "--append-ini", "{}={}".format(ini_option_name, append_value), "-s"
+        )
+        result.stdout.fnmatch_lines(stdout_match)
+        result.stderr.fnmatch_lines("WARNING: {}".format(stderr_match))
+
+        result = testdir.runpytest(
+            "--append-ini",
+            "{}={}".format(ini_option_name, append_value),
+            "-s",
+            "--strict-config",
+        )
+        failures_text = [
+            "=*= FAILURES =*=",
+            stderr_match,
+        ]
+        result.stdout.fnmatch_lines(failures_text)
+
+    def test_append_ini_usage_no_such_key(self, testdir):
+        testdir.makeconftest(
+            """
+            def pytest_addoption(parser):
+                addini = parser.addini
+                addini("my_args", "", default="foo", type="args")"""
+        )
+        testdir.makepyfile(
+            """
+            def test_pass(pytestconfig):
+                ini_val = pytestconfig.getini("my_args")
+                print('\\nappended_option:%s\\n' % ini_val)"""
```

Comment:
Using a raw string can avoid the escapes which I think looks a bit nicer:

```suggestion
            r"""
            def test_pass(pytestconfig):
                ini_val = pytestconfig.getini("my_args")
                print('\nappended_option:%s\n' % ini_val)"""
```